apps/api/src/app/events/e2e/bridge-trigger.e2e.ts

```diff
@@ -949,6 +961,508 @@ contexts.forEach((context: Context) => {
       });
 
       expect(sentMessages.length).to.be.eq(0);
+
+      const executionDetailsFiltered = await executionDetailsRepository.find({
+        _environmentId: session.environment._id,
+        status: ExecutionDetailsStatusEnum.SUCCESS,
+      });
+
+      const executionDetailsWorkflowFiltered = executionDetailsFiltered.filter(
+        (executionDetail) => executionDetail.detail === DetailEnum.STEP_FILTERED_BY_WORKFLOW_RESOURCE_PREFERENCES
+      );
+
+      expect(executionDetailsWorkflowFiltered.length).to.be.eq(1);
+    });
+
+    it(`should deliver inApp message if subscriber disabled inApp channel for readOnly workflow with inApp enabled [${context.name}]`, async () => {
+      process.env.IS_WORKFLOW_PREFERENCES_ENABLED = 'true';
+      const workflowId = `enabled-readonly-workflow-level-${`${context.name}-${uuidv4()}`}`;
+      const newWorkflow = workflow(
+        workflowId,
+        async ({ step }) => {
+          await step.inApp('send-in-app', () => ({ body: 'Hello there 1' }));
+        },
+        {
+          preferences: {
+            all: {
+              readOnly: true,
+            },
+            channels: {
+              inApp: {
+                enabled: true,
+              },
+            },
+          },
+        }
+      );
+
+      await bridgeServer.start({ workflows: [newWorkflow] });
+
+      if (context.isStateful) {
+        await discoverAndSyncBridge(session, workflowsRepository, workflowId, bridgeServer);
+      }
+
+      const createdWorkflow = await workflowsRepository.findByTriggerIdentifier(session.environment._id, workflowId);
+
+      if (context.isStateful) {
+        // Set subscriber preference to disable inApp for the workflow
+        await session.testAgent
+          .patch(`/v1/inbox/preferences/${createdWorkflow?._id}`)
+          .set('Authorization', `Bearer ${session.subscriberToken}`)
+          .send({
+            in_app: false,
+          });
+      }
+
+      await triggerEvent(session, workflowId, subscriber._id, {}, bridge);
```

Comment:
Using `subscriber._id` instead of `subscriber.subscriberId` per the explanation above.